packages/flutter/lib/src/widgets/web_image.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../widgets.dart';
+import '_web_image_io.dart' if (dart.library.js_interop) '_web_image_web.dart';
+
+/// A widget that displays an image from the web.
+///
+/// Due to restrictions on cross-origin images on the web, in cases where the
+/// image [url] is cross-origin and access to the byte data is disallowed, this
+/// widget will instead use a platform view to display the image in an `<img>`
+/// element.
+///
+/// If the [src] of the image is same-origin, or if we are able to make a
+/// cross-origin request for the image bytes, then this widget works exactly
+/// the same as [Image.network].
+class WebImage extends StatefulWidget {
```

Comment:
I wonder about the discoverability of this new widget. I'm not sure if we could have a named constructor on `Image` instead? (that no-op's on native platforms; `Image.file` does the opposite, for web).

If this is possible, then we could get ImageCache integration out-of-the-box?

Reply:
The main problem with that approach is that `Image` widgets expose `ImageProvider`s which allow access to the underlying `ui.Image`. But `WebImage`, if backed by an `<img>` element, cannot create a `ui.Image` object since that would require access to the underlying bytes. Due to security restrictions on the web, the `WebImage` can't be as fully-featured as the standard `Image` widget (for example, you don't have control over how it shows multi-frame images).

Reply:
We did think a bit about making the `Image.network` constructor forward to an `<img>`-backed Widget on the web in cases where the `url` was cross-origin, but the problem there is, again, we aren't creating a `ui.Image`.

Reply:
Maybe we could add a code sample that uses `kIsWeb` to decide whether to use one or the other?
That would be a good compromise, I think.

Reply:
We use `Image.network` now.